null_int.go

```diff
@@ -0,0 +1,103 @@
+package dosa
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+)
+
+// NullInt is a type that holds an optional int64 value.
```

Comment:
database/sql is part of the go core, and we're only pulling in the types, so it's not a big deal. However, if you still feel strongly about making new types, I'm okay with that too.

I don't know that we need Scan, so if you want to make your own types please make the structures as close to the ones in database/sql as possible (same names and methods).

Otherwise looks good.